python/Homework.py

```diff
@@ -0,0 +1,210 @@
+import numpy as np
+import pandas as pd
+import plotly.express as px
+from sklearn import tree
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler
+
+# 1 Load the above data into a Pandas DataFrame
+heads = (
+    "sepal length in cm",
+    "sepal width in cm",
+    "petal length in cm",
+    "petal width in cm",
+    "class",
+)
+Iris_df = pd.read_csv(
+    "https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data",
+    names=heads,
+)
+Iris_df1 = pd.read_csv(
+    "https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data"
+)
+
+nd_array_1 = np.array(Iris_df1)[:, 0:4]
+
+# 2-1 Mean
+print(nd_array_1.mean(axis=0))
+
+# 2-2 Min and Max
+print(nd_array_1.min(axis=0))
+print(nd_array_1.max(axis=0))
+
+# 2-3 Quartiles
+Lower = np.quantile(nd_array_1, 0.25, axis=0, interpolation="lower")
+Higher = np.quantile(nd_array_1, 0.75, axis=0, interpolation="higher")
+print(Lower)
+print(Higher)
+
+# 2-3-1 The other way for Quartile
+Percent = np.percentile(nd_array_1, [25, 50, 75], axis=0)
+print(Percent)
+
+# 3-1 Scatter Plot
+
+fig = px.scatter(
+    Iris_df,
+    x="sepal length in cm",
+    y="sepal width in cm",
+    color="class",
+    size="petal width in cm",
+    hover_data=["petal width in cm"],
+    symbol="class",
+)
+fig.show()
+
+# From the scatter plot, Senoto has higher sepal width but lower sepal length compare to other two kinds
+# Scatter plot is easy to tell the difference if the difference is obvious among the kinds.
+
+# 3-2 Violin Plot
+
+df = px.data.tips()
```

Comment:
I don't think we need the tips data here. Could you please check?

Reply:
No, we do not use it so I delete it!

Reply:
Nice catch!